blog/content/post/2025-07-xx-edk2-on-odroid-m2.md

```diff
@@ -0,0 +1,325 @@
+---
+title: Booting EDK2 on Odroid M2
+abstract: ''
+cover: /img/tianocore_logo.jpg
+author: michal.kopec
+layout: post
+published: true
+date: 2025-07-01
+archives: "2025"
+
+tags:
+  - QubesOS
+categories:
+  - Firmware
+
+---
+
+This post describes the process of booting TianoCore EDK II on an Odroid M2
+platform (ARM64, RK3588 based) and gives an overview of the status of ARM
+platforms in EDK II.
+
+## Introduction
+
+TianoCore EDK II is the reference implementation of UEFI and is the de facto
+standard boot firmware in the x86 space. In the ARM world, especially in the
+embedded Linux space, that role is typically served by U-boot. With the rise of
+high performance ARM computing, however, EDK II is becoming an increasingly
+popular option for system builders. In this blog post I'll be checking out the
+process of booting EDK II and an UEFI OS on an Arm platform with pre-existing
+SoC support.
+
+## The platform
+
+![img](/img/odroid_m2_board.jpg)
+
+I picked an Odroid M2, which is an SBC with a Rockchip RK3588 SoC. I wanted
+that SoC in particular, because there is already some open-source EDK II support
+for it, in the form of [edk2-porting/edk2-rk3588](https://github.com/edk2-porting/edk2-rk3588).
+
+The M2 is a pretty neat little board that can run Ubuntu 25.04 smoothly, has an
+NPU for users who want it, and most of the Linux support code is already merged.
```

Comment:
@m-iwanicki we plan to have during ZDM#2 or probably more likely ZDM#3 some discussion about NPU on that hardware, if there would be a CfP submitted, I could make it public and we could co-promote your talk.